tools/pd-ctl/pdctl/command/store_command.go

```diff
@@ -304,54 +327,124 @@ func deleteStoreCommandFunc(cmd *cobra.Command, args []string) {
 }
 
 func deleteStoreCommandByAddrFunc(cmd *cobra.Command, args []string) {
+	id := getStoreID(cmd, args, false)
+	if id == 0 {
+		return
+	}
+	// delete store by its ID
+	prefix := fmt.Sprintf(storePrefix, id)
+	_, err := doRequest(cmd, prefix, http.MethodDelete)
+	if err != nil {
+		cmd.Printf("Failed to delete store %s: %s\n", args[0], err)
+		return
+	}
+	cmd.Println("Success!")
+}
+
+func cancelDeleteStoreCommandFunc(cmd *cobra.Command, args []string) {
 	if len(args) != 1 {
 		cmd.Usage()
 		return
 	}
+	if _, err := strconv.Atoi(args[0]); err != nil {
+		cmd.Println("store_id should be a number")
+		return
+	}
+
+	prefix := fmt.Sprintf(storePrefix, args[0])
+	r, err := doRequest(cmd, prefix, http.MethodGet)
+	if err != nil {
+		cmd.Printf("Failed to get store: %s\n", err)
+		return
+	}
+
+	storeInfo := struct {
```

Comment:
Can the defined struct be used here?

Reply:
We can but I don't think it's necessary.